pkg/actions/nodegroup/upgrade.go

```diff
@@ -373,7 +377,7 @@ func (m *Manager) usesCustomAMIEKSNodeGroup(ctx context.Context, ng *ekstypes.No
 	}
 
 	return m.usesCustomAMI(ctx, &api.LaunchTemplate{
-		ID:      *ng.LaunchTemplate.Id,
+		ID:      ng.LaunchTemplate.Id,
 		Version: ng.LaunchTemplate.Version,
 	})
 }
```

Comment:
I believe the changes in this file are not necessary. Here we first retrieve the launch template for the nodegroup's cloudformation stack, so the resource we get back should always have an ID. Checking for name is redundant in this workflow.